controller/engine_controller.go

```diff
@@ -1301,6 +1301,15 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 		ec.backoff.DeleteEntry(e.Name)
 		ec.eventRecorder.Eventf(e, v1.EventTypeNormal, EventReasonRebuilded,
 			"Replica %v with Address %v has been rebuilded for volume %v", replica, addr, e.Spec.VolumeName)
+
+		// Call snapshotpurge to clean up system generated snapshot.
+		if err := client.SnapshotPurge(); err != nil {
+			log.WithError(err).Errorf("Failed to start snapshot purge for volume %v", e.Spec.VolumeName)
+			ec.eventRecorder.Eventf(e, v1.EventTypeWarning, EventReasonFailedStartingSnapshotPurge, "Failed to start snapshot purge for volume %v: %v", e.Spec.VolumeName, err)
+		} else {
+			logrus.Debugf("Started snapshot purge for volume %v after rebuild is done", e.Spec.VolumeName)
+			ec.eventRecorder.Eventf(e, v1.EventTypeNormal, EventReasonStartedSnapshotPurge, "Started snapshot purge for volume %v after rebuild is done", e.Spec.VolumeName)
```

Comment:
Don't need to send an event for this since it's the common path.

Reply:
ok, will fix it. Thanks.